src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/InitiatePaymentTest.java

```diff
@@ -0,0 +1,162 @@
+import uk.gov.hmcts.cmc.claimstore.BaseSaveTest;
+import uk.gov.hmcts.cmc.domain.models.ClaimData;
+import uk.gov.hmcts.cmc.domain.models.ioc.InitiatePaymentResponse;
+import uk.gov.hmcts.cmc.domain.models.sampledata.SampleClaimData;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.cmc.ccd.domain.CaseEvent.INITIATE_CLAIM_PAYMENT_CITIZEN;
+import static uk.gov.hmcts.cmc.claimstore.utils.ResourceLoader.successfulCoreCaseDataStoreStartResponse;
+import static uk.gov.hmcts.cmc.claimstore.utils.ResourceLoader.successfulCoreCaseDataStoreSubmitResponse;
+
+@TestPropertySource(
+    properties = {
+        "document_management.url=false",
+        "feature_toggles.ccd_async_enabled=false",
+        "feature_toggles.ccd_enabled=true",
+        "feature_toggles.async_event_operations_enabled=false"
+    }
+)
+public class InitiatePaymentTest extends BaseSaveTest {
+
+    @Test
+    public void shouldReturnNewlyCreatedClaim() throws Exception {
+        ClaimData claimData = SampleClaimData.submittedByClaimant();
+
+        given(coreCaseDataApi.startForCitizen(
+            eq(AUTHORISATION_TOKEN),
+            eq(SERVICE_TOKEN),
+            eq(USER_ID),
+            eq(JURISDICTION_ID),
+            eq(CASE_TYPE_ID),
+            eq(INITIATE_CLAIM_PAYMENT_CITIZEN.getValue())
+            )
+        ).willReturn(successfulCoreCaseDataStoreStartResponse());
+
+        given(coreCaseDataApi.submitForCitizen(
+            eq(AUTHORISATION_TOKEN),
+            eq(SERVICE_TOKEN),
+            eq(USER_ID),
+            eq(JURISDICTION_ID),
+            eq(CASE_TYPE_ID),
+            eq(IGNORE_WARNING),
+            any()
+            )
+        ).willReturn(successfulCoreCaseDataStoreSubmitResponse());
+
+        given(authTokenGenerator.generate()).willReturn(SERVICE_TOKEN);
+
+        MvcResult result = makeInitiatePaymentRequest(claimData, AUTHORISATION_TOKEN)
+            .andExpect(status().isOk())
+            .andReturn();
+
+        verify(coreCaseDataApi)
+            .startForCitizen(
+                eq(AUTHORISATION_TOKEN),
+                eq(SERVICE_TOKEN),
+                eq(USER_ID),
+                eq(JURISDICTION_ID),
+                eq(CASE_TYPE_ID),
+                eq(INITIATE_CLAIM_PAYMENT_CITIZEN.getValue())
+            );
+
+        verify(coreCaseDataApi)
+            .submitForCitizen(
+                eq(AUTHORISATION_TOKEN),
+                eq(SERVICE_TOKEN),
+                eq(USER_ID),
+                eq(JURISDICTION_ID),
+                eq(CASE_TYPE_ID),
+                eq(IGNORE_WARNING),
+                any()
```

Comment:
Not sure if my understanding is right, should we can verify on the specific object rather than any()?

Reply:
The assertion is down, we're deserialising the response and checking it is what we expect, so in this case there's no need to be detailed on `verify`. This is an endpoint test - I could've removed the verify all together and the test would still have merit.